### PR TITLE
Add atreides05

### PR DESCRIPTION
**Pull request by abcdefg30**
Depends on #12784.
I also still need to get the colors for Smugglers and Mercenaries (which I'll do tomorrow).

### CODE REVIEW

**Review by Micr0Bit — APPROVED** [review]
theres one thing i dont like ... that units dive into the shroud on top and bottom edge of the map

but i assume that has nothing to do with the mission itself ? 

regardless, i finished the mission without any technical problems :+1:

**Comment by MustaphaTR** [comment]
I used DDDD00 for mercenaries and 542209 for smugglers in my older maps (and Ord4 for smugglers too).

File: mods/d2k/maps/atreides-05/atreides05-AI.lua
Comment (MustaphaTR) [inline mods/d2k/maps/atreides-05/atreides05-AI.lua]: AI doesn't have an IX Lab in this mission. So it shouldn't be able to build Missile Tank.

**Review by MustaphaTR — CHANGES_REQUESTED** [review]

**Comment by MustaphaTR** [comment]
We can't build combat tank from Starport.

Also i like the fact that Contrabants contain Ordos Tanks not Atreides. This wasn't the case in original even tho it was possible.

I feel like it would be better if Combat Tank we can buy from Smugglers was Ordos too, it is a Smugglers Starport. We can still build atreides ones from HFac. But that's just my idea.

File: mods/d2k/maps/atreides-05/atreides05.lua
Comment (MustaphaTR) [inline mods/d2k/maps/atreides-05/atreides05.lua]: Mercenaries attack the starport base first, if there is still a base, in original game, not main base.

File: mods/d2k/maps/atreides-05/atreides05-AI.lua
Comment (abcdefg30) [inline mods/d2k/maps/atreides-05/atreides05-AI.lua]: Oh, ops. Thanks fixed.

**Comment by abcdefg30** [comment]
> I feel like it would be better if Combat Tank we can buy from Smugglers was Ordos too, it is a Smugglers Starport. We can still build atreides ones from HFac. But that's just my idea.

Good idea. I added that.

> I used DDDD00 for mercenaries and 542209 for smugglers in my older maps (and Ord4 for smugglers too).

Also thanks. I think they fit.

File: mods/d2k/maps/atreides-05/rules.yaml
Comment (MustaphaTR) [inline mods/d2k/maps/atreides-05/rules.yaml]: Frigate is misspelt on comment.

**Comment by MustaphaTR** [comment]
Needs rebase.

**Comment by MustaphaTR** [comment]
This walls and gun turret is lacking concrete below:
concrete

**Comment by abcdefg30** [comment]
Rebased and updated.

**Comment by MustaphaTR** [comment]
AI in original mission do train infantry. I tested with Debug Menu in Gruntmods Edition. Looks like it is missing here. Don't forget to add `upgrade.barracks` too.

**Comment by MustaphaTR** [comment]
If i sell starport, Defend Starport mission doesn't fail.

**Comment by MustaphaTR** [comment]
Sardaukar is visible and buildable if you have hi-tech fac. after i capture Harkonnen Barracks/capture the CYard and build a Barracks.

**Comment by HenrytheSlav** [comment]
Please add the .lua files to [OpenRA.sln](https://github.com/OpenRA/OpenRA/blob/bleed/OpenRA.sln#L101).

**Comment by reaperrr** [comment]
#12784 was merged.

**Comment by abcdefg30** [comment]
Rebased and updated.

**Review by MustaphaTR — APPROVED** [review]
:+1:

**Review by ltem — APPROVED** [review]
:+1: Quite a challenge :)
One small nit...

File: mods/d2k/maps/atreides-05/atreides05.lua
Comment (ltem) [inline mods/d2k/maps/atreides-05/atreides05.lua]: A line break would be nice here

**Comment by abcdefg30** [comment]
Rebased and updated.

**Review by ltem — APPROVED** [review]

**Review by MustaphaTR — APPROVED** [review]

**Comment by MustaphaTR** [comment]
@reaperrr , can you take a look to this when you have time. It already have 2 approvals but waiting for merge for a while.

**Review by reaperrr — APPROVED** [review]

**Comment by reaperrr** [comment]
[changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/dedef53bdb8c57553a81d1f9eff055ddd7388494...d067edd38d9db0c63dd56f7609c49eb18538117f)

**Pull request merged**
